maintenance_product/models/maintenance_equipment.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class MaintenanceEquipment(models.Model):
+    _inherit = "maintenance.equipment"
+
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        string="Product",
+        tracking=True,
+        domain=[
+            ("categ_id", "child_of", "product_category_id"),
+            ("maintenance_ok", "=", True),
+        ],
```

Comment:
@BhaveshHeliconia 
As the domain is structured now, it will always return the string 'product_category_id' instead of the actual ID.

You probably want something like:
`"[('categ_id', 'child_of', product_category_id), ('maintenance_ok', '=', True)]"`

so that product_category_id is evaluated properly.